src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2303,6 +2303,35 @@ private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byt
             return false;
         }
 
+#pragma warning disable CA1822 // Mark members as static
+        private bool readObject(CORINFO_OBJECT_STRUCT_* objPtr, byte* buffer, int bufferSize, int valueOffset)
```

Comment:
Is this exercised by any tests?

Reply:
Good point, it definitely is for JIT, not sure about NAOT (e.g. because of https://github.com/dotnet/runtime/issues/83043)

I'll run the extras pipeline on CI to check that if you don't mind (not near the computer atm) - I've pushed an intentional bug in the impl

Reply:
Added a test and verified that it fails with the "intentional buggy impl"